### PR TITLE
fix: fix binary search crash in minhash_lsh index

### PR DESCRIPTION
**Pull request by cqy123456**
issue : https://github.com/zilliztech/knowhere/issues/1320

### CODE REVIEW

File: src/simd/distances_avx512.cc [hunk 1]
```diff
@@ -914,70 +914,132 @@ rabitq_dp_popcnt_avx512(const uint8_t* q, const uint8_t* x, const size_t d, cons
...
+
     return -1;
 }
 int
 u64_binary_search_ge_avx512(const uint64_t* data, const size_t size, const uint64_t target) {
-    constexpr int SIMD_WIDTH = 8;
+    if (size == 0)
+        return -1;
+
+    if (size < 64) {
```
Comment (alexanderguzhva) [inline src/simd/distances_avx512.cc]: the previous condition in the function above was `if (size <= 64)`. I just want to confirm that this is intentional

File: tests/ut/test_simd.cc [hunk 2]
```diff
@@ -47,14 +47,159 @@ TEST_CASE("Test minhash function") {
     auto simd_type = knowhere::KnowhereConfig::SimdType::AVX512;
     knowhere::KnowhereConfig::SetSimdType(simd_type);
     constexpr size_t seed = 111;
-    SECTION("test binary search funtion ") {
-        size_t size = 1200;
-        auto x = GenRandomVector<uint64_t>(size, 1, seed);
+    SECTION("test binary search function with comprehensive corner cases") {
+        auto size =
+            GENERATE(as<size_t>{}, 1, 2, 4, 7, 8, 12, 14, 16, 21, 28, 32, 35, 42, 49, 56, 64, 111, 128, 256, 1000);
+        auto x = GenRandomVector<uint64_t>(size, 1, seed, 10000);
```
Comment (alexanderguzhva) [inline tests/ut/test_simd.cc]: please do a verification for several random vectors, just add several (5?) iterations of generating `x` and testing it

File: src/simd/distances_avx512.cc (hunk 1 above)
Comment (cqy123456) [reply src/simd/distances_avx512.cc]: You're right. This code is redundant. If the size is less than 32,  jump to the non-SIMD path

File: tests/ut/test_simd.cc (hunk 2 above)
Comment (cqy123456) [reply tests/ut/test_simd.cc]: updated

**Comment by foxspy** [comment]
analyzer should be fixed @cqy123456

File: src/index/minhash/minhash_index_node.cc [hunk 3]
```diff
@@ -309,7 +309,7 @@ MinHashLSHNode<DataType>::Search(const DataSetPtr dataset, std::unique_ptr<Confi
             std::vector<folly::Future<folly::Unit>> futures;
             constexpr size_t batch_size = 64;
             size_t run_time = (nq + batch_size - 1) / batch_size;
```
Comment (foxspy) [inline src/index/minhash/minhash_index_node.cc]: run_times

File: src/index/minhash/minhash_lsh.h [hunk 4]
```diff
@@ -581,8 +581,8 @@ MinHashLSH::BatchSearch(const char* query, size_t nq, float* distances, idx_t* l
                                               this->mh_vec_elememt_size_, refine_k, topk, res_dis, res_ids);
                 return;
             }));
-            WaitAllSuccess(futures);
         }
+        WaitAllSuccess(futures);
```
Comment (foxspy) [inline src/index/minhash/minhash_lsh.h:585]: Was it running serially before...?

**Review by foxspy — APPROVED** [review]
/lgtm

File: src/index/minhash/minhash_lsh.h (hunk 4 above)
Comment (alexanderguzhva) [reply src/index/minhash/minhash_lsh.h:585]: I hope it has not been running in N^2 mode, because the `futures` got never cleared and I hope that finished `future` object would not re-run again

Comment (alexanderguzhva) [reply src/index/minhash/minhash_lsh.h:585]: and it is great that the performance can be improved quite a lot, it seems!
